dozer-ingestion/src/connectors/postgres/helper.rs

```diff
@@ -147,15 +180,30 @@ pub fn value_to_field(
             let value: Result<Vec<u8>, _> = row.try_get(idx);
             value.map_or_else(handle_error, |v| Ok(Field::Binary(v)))
         }
-        &Type::JSONB => {
+        &Type::JSONB | &Type::JSON => {
             let value: Result<serde_json::Value, _> = row.try_get(idx);
-
-            value.map_or_else(handle_error, |v| {
-                Ok(Field::Bson(
-                    bson::to_vec(&v).map_err(|e| JSONBParseError(e.to_string()))?,
+            value.map_or_else(handle_error, |val| {
+                Ok(Field::Json(
+                    serde_json_to_json_value(val)
+                        .map_err(|_| ColumnTypeNotSupported(col_type.name().to_string()))
+                        .unwrap(),
                 ))
             })
         }
+        &Type::JSONB_ARRAY | &Type::JSON_ARRAY => {
+            let value: Result<Vec<serde_json::Value>, _> = row.try_get(idx);
+            value.map_or_else(handle_error, |val| {
+                Ok(Field::Json(JsonValue::Array(
+                    val.into_iter()
+                        .map(|v| {
+                            serde_json_to_json_value(v)
+                                .map_err(|_| ColumnTypeNotSupported(col_type.name().to_string()))
```

Comment:
This is not ColumnTypeNotSupported, maybe we can forward the TypeError?

Reply:
yup!